pkg/config/ext_authz.go

```diff
@@ -88,17 +114,28 @@ func init() {
 
 	ExternalAuthz = &ExternalAuthzConfig{
 		Enabled:    true,
-		Cluster:    extAuthzCluster(host, uint32(port)),
-		HTTPFilter: externalAuthZFilter(extAuthzClusterName, timeout, env.FailureModeAllow, env.MaxRequestBytes),
+		Cluster:    extAuthzCluster(host, uint32(port), env.Protocol),
+		HTTPFilter: externalAuthZFilter(extAuthzClusterName, env.Host, env.PathPrefix, timeout, env.FailureModeAllow, env.MaxRequestBytes, env.Protocol),
```

Comment:
I would pass `env` to the args instead instead of five `env.*` variables by changing `externalAuthZFilter()` to accept `*config`.
```suggestion
		HTTPFilter: externalAuthZFilter(extAuthzClusterName, timeout, env),
```